src/validators/rdf.js

```diff
@@ -0,0 +1,105 @@
+import XMLDom from 'xmldom';
+
+import { RDF_OBSOLETE_TAGS, RDF_UNALLOWED_TAGS, RDF_UNALLOWED_IF_LISTED_TAGS,
+  VALIDATION_ERROR, VALIDATION_WARNING } from 'const';
+import { RDFParseError } from 'exceptions';
+import * as messages from 'messages';
+
+
+export default class RDFScanner {
+
+  constructor(contents, filename) {
+    this.contents = contents;
+    this.filename = filename;
+    // I don't think this ever needs to be different, but if it does we can
+    // extract the em namespace using:
+    //
+    // this.namespace = this._xmlDoc.documentElement._nsMap.em;
+    this.namespace = 'http://www.mozilla.org/2004/em-rdf#';
+    this.validatorMessages = [];
+    this._xmlDoc = null;
+  }
+
+  scan() {
+    return new Promise((resolve, reject) => {
+      this.mustNotExist()
+        .then(() => {
+          resolve(this.validatorMessages);
+        })
+        .catch(reject);
+    });
+  }
+
+  getXMLDoc() {
+    return new Promise((resolve, reject) => {
+      if (this._xmlDoc !== null) {
+        return resolve(this._xmlDoc);
+      }
+
+      this._xmlDoc = new XMLDom.DOMParser({
+        errorHandler: (err) => {
+          reject(new RDFParseError(`RDF Parse Error: ${err}`));
+        },
+      }).parseFromString(this.contents);
+
+      resolve(this._xmlDoc);
+    });
+  }
+
+  mustNotExist() {
+    return new Promise((resolve, reject) => {
+      this.getXMLDoc()
+        .then((xmlDoc) => {
+          var bannedTags = RDF_UNALLOWED_TAGS;
+          var addonIsListed = (xmlDoc
+            .getElementsByTagNameNS(this.namespace, 'listed').length > 0);
+
+          if (addonIsListed) {
```

Comment:
Thanks; turns out I wasn't even exporting a message for this one! Yay for testing.